action_server/src/robocorp/action_server/_server_expose.py

```diff
@@ -98,23 +118,74 @@ def forward_request(base_url: str, payload: BodyPayload) -> requests.Response:
 async def handle_ping_pong(
     ws: websockets.WebSocketClientProtocol,
     pong_queue: asyncio.Queue,
-    no_connection_queue: asyncio.Queue,
+    ping_interval: int,
 ):
     while True:
-        await asyncio.sleep(2)
+        await asyncio.sleep(ping_interval)
         await ws.send("ping")
 
         try:
-            await asyncio.wait_for(pong_queue.get(), timeout=2)
-        except asyncio.TimeoutError:
-            log.debug("Ping-pong message timeout")
+            await asyncio.wait_for(pong_queue.get(), timeout=ping_interval)
+        except (asyncio.TimeoutError, TimeoutError):
+            log.debug("Lost connection to expose server")
             await ws.close()
-            await no_connection_queue.put("no_connection")
             break
 
 
-class ExposePingTimeoutError(Exception):
-    pass
+def handle_session_payload(
+    session_payload: SessionPayload, api_key: str | None, expose_url: str, datadir: str
+):
+    url = f"https://{session_payload.sessionId}.{expose_url}"
+    log.info(f"🌍 URL: {url}")
+    if api_key is not None:
+        log.info(
```

Comment:
Question: shouldn't the api_key be checked here?

Reply:
Not relevant after https://github.com/robocorp/robocorp/pull/161